common/web/gesture-recognizer/src/tools/unit-test-resources/src/touchpathTurtle.ts

```diff
@@ -3,11 +3,15 @@ namespace Testing {
   type InputSample     = com.keyman.osk.InputSample;
   type Subsegmentation = com.keyman.osk.Subsegmentation;
 
+  interface EventMap {
+    'sample': (sample: InputSample) => void
+  }
+
   /**
    * Designed to facilitate creation of 'synthetic' touchpath sample sequences for use
    * in unit tests.
    */
-  export class TouchpathTurtle {
+  export class TouchpathTurtle extends EventEmitter<EventMap> {
```

Comment:
This allows the class to serve as a source of synthetic touch events for non-segmentation unit tests.  It'll definitely come in handy for easily-maintainable synthetic touchpath sequences for unit tests, as seen with the new unit test "synthetic sequence: start -> hold -> move -> end".